torch/_dynamo/variables/nn_module.py

```diff
@@ -211,35 +218,19 @@ def record_nn_module_stack():
                 )
 
             else:
-                # for lazy modules, run the pre-hooks which will update the type
-                # TODO mlazos: we don't fully support all of the hooks that exist,
-                # so restrict using __call__ only to lazy modules for now
                 assert self.source, (
                     "Must provide a valid source in order to inline, "
                     "since inlined function may have default args which must be guarded."
                 )
-                if is_lazy:
```

Comment:
Does removing this trigger a recompile on lazy modules?

Reply:
i didn't check that. (would any existing test cover that case?)

but i don't think so, because i should be making the behavior that used to apply to lazy modules now apply more widely.

Reply:
Might be worth running a simple example and checking how many compiles you get

Reply:
I confirmed the behavior is the same on master and on my PR for a simple case I based off of the lazy module in `test/dynamo/test_modules.py` and added compile counts in more places.

My observation was that we do not capture 1 graph for the first try in either specialized or unspecialized case, but, we also do not recompile on a second run of either.  Does that address your concern?